sdk/java/core/src/main/kotlin/com/keepersecurity/secretsManager/core/SecretsManager.kt

```diff
@@ -777,46 +797,45 @@ private fun fetchAndDecryptSecrets(
 }
 
 @ExperimentalSerializationApi
-private fun decryptRecord(record: SecretsManagerResponseRecord, recordKey: ByteArray): KeeperRecord? {
-    val decryptedRecord = decrypt(record.data, recordKey)
-
-    val files: MutableList<KeeperFile> = mutableListOf()
-
-    if (record.files != null) {
-        record.files.forEach {
-            val fileKey = decrypt(it.fileKey, recordKey)
-            val decryptedFile = decrypt(it.data, fileKey)
-            files.add(
-                KeeperFile(
-                    fileKey,
-                    it.fileUid,
-                    Json.decodeFromString(bytesToString(decryptedFile)),
-                    it.url,
-                    it.thumbnailUrl
-                )
-            )
-        }
-    }
-
-    // When SDK is behind/ahead of record/field type definitions then
-    // strict mapping between JSON attributes and object properties
-    // will fail on any unknown field/key - currently just log the error
-    // and continue without the field - nb! field will be lost on save
-    var recordData: KeeperRecordData? = null
-    try {
-        recordData = Json.decodeFromString<KeeperRecordData>(bytesToString(decryptedRecord))
-    } catch (e: Exception) {
-        // New/missing field: Polymorphic serializer was not found for class discriminator 'UNKNOWN'...
-        // New/missing field property (field def updated): Encountered unknown key 'UNKNOWN'.
-        // Avoid 'ignoreUnknownKeys = true' to prevent erasing new properties on save/update
-        println("Record ${record.recordUid} has unexpected data properties (ignored).\n"+
-                " Error parsing record type - KSM SDK is behind/ahead of record/field type definitions." +
-                " Please upgrade to latest version. If you need assistance please email support@keepersecurity.com")
-        //println(e.message)
-        recordData = nonStrictJson.decodeFromString<KeeperRecordData>(bytesToString(decryptedRecord))
+private fun decryptRecord(encryptedRecord: SecretsManagerResponseRecord, recordKey: ByteArray): Result<KeeperRecord> {
+
+    val decryptedRecord = decrypt(encryptedRecord.data, recordKey)
+
+    val decryptedRecordFiles = encryptedRecord.files?.map { recordFile ->
+        val fileKey = decrypt(recordFile.fileKey, recordKey)
+        val decryptedFile = decrypt(recordFile.data, fileKey)
+        KeeperFile(
+            fileKey,
+            recordFile.fileUid,
+            Json.decodeFromString(bytesToString(decryptedFile)),
+            recordFile.url,
+            recordFile.thumbnailUrl
+        )
+    } ?: emptyList()
+
+    val recordData: KeeperRecordData = try {
```

Comment:
Restructuring this ensures that there are only two  non exception pathways out - either you get the record data you want, or you don't, and both are ok.